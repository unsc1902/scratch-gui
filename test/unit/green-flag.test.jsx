/* eslint-env jest */
const React = require('react');
const {shallow} = require('enzyme');
const GreenFlag = require('../../src/containers/green-flag');
const renderer = require('react-test-renderer');
const VM = require('scratch-vm');

describe('GreenFlag', () => {
    let vm;
    beforeEach(() => {
        // TODO(bogusred): Ask scratch team their thoughts on mocking the VM.
        vm = new VM();
    });

    test('renders active state', () => {
        const component = renderer.create(
            <GreenFlag active={true} vm={vm}/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('renders inactive state', () => {
        const component = renderer.create(
            <GreenFlag active={false} vm={vm}/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('triggers onClick when active', () => {
        const onClick = jest.fn();
        const componentShallowWrapper = shallow(
            <GreenFlag active={true} onClick={onClick} vm={vm}/>
        );
        componentShallowWrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });

    // TODO(bogusred): Test for handles key events.
    // TODO(bogusred): Test project run start.
    // TODO(bogusred): Test project run stop.
});
