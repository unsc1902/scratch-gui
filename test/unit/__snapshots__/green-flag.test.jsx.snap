// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenFlag renders active state 1`] = `
<img
  className="undefined"
  onClick={[Function]}
  src="test-file-stub"
  title="Go"
/>
`;

exports[`GreenFlag renders inactive state 1`] = `
<img
  className=""
  onClick={[Function]}
  src="test-file-stub"
  title="Go"
/>
`;
